src/dlr_tvm.cc

```diff
@@ -198,3 +207,90 @@ void TVMModel::UseCPUAffinity(bool use) {
     LOG(INFO) << "CPU Affinity is disabled";
   }
 }
+
+bool TVMModel::HasMetadata() const {
+  if (this->metadata != nullptr) {
+    return true;
+  }
+  return false;
+}
+
+void TVMModel::GetOutputNames(char**& names) const {
+  names = (char**) malloc(this->num_outputs_);
+  std::string name;
+
+  for(int i = 0; i < this->num_outputs_; i++) {
+    name = this->metadata["Model"]["Outputs"][i]["name"];
+    names[i] = (char *)malloc((name.size() + 1)*sizeof(char));
```

Comment:
Lets not do malloc each time when this method is called. We can do the following. TVMModel internally has `vector<string> input_names` storing input names. Lets create another vector in TVMModel - `vector<const char*> c_input_names` which stores pointers to c_strings of input_names vectors. When user calls `GetOutputNames` we simply give user `c_input_names` pointer. Example: https://gist.github.com/apivovarov/10be0d6a12dfc294ac87ee287e70deab
Benefits of that solution that we do not call `malloc` on every single method call. Users do not need to do memory cleanup. Price for that solution is new vector which stores 64 bit addresses of actual input names - very cheap.

Reply:
Cool. Nice thanks